### PR TITLE
Reserve segments of dynamic large object when delete objects in container what was enabled versioning.

### PR DESCRIPTION
**Pull request by nguyenhuuluan434**
Hi @ncw,
Please help me review the change what i add code to handle reserve sgements if objects in the container what was enabled versioning.

Thank in advance,
Luan Nguyen Huu.

### CODE REVIEW

File: backend/swift/swift.go [hunk 1]
```diff
@@ -530,10 +530,10 @@ type listFn func(remote string, object *swift.Object, isDirectory bool) error
 //
 // Set recurse to read sub directories
 func (f *Fs) listContainerRoot(container, directory, prefix string, addContainer bool, recurse bool, fn listFn) error {
-	if prefix != "" {
```
Comment (ncw) [inline backend/swift/swift.go]: This looks like an unrelated change?  What is it for?

File: backend/swift/swift.go [hunk 2]
```diff
@@ -1306,6 +1330,10 @@ func (o *Object) Remove(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	isInContainerVersioning, err := o.isInContainerVersioning()
```
[the remark this answers is not in the document]
Comment (nguyenhuuluan434) [reply backend/swift/swift.go]: Yes, i will check `isInContainerVersioning` when condtion `isDynamicLargeObject` is true.

**Comment by ncw** [comment]
That looks great now - I'll merge it - thank you :-)

**Comment by ncw** [comment]
I think this test might have broken the integration tests

See swift failures here: https://pub.rclone.org/integration-tests/current/

and: https://pub.rclone.org/integration-tests/current/swift-backend.swift-TestSwift-5.txt

Any ideas?

**Comment by ncw** [comment]
That URL should be: https://pub.rclone.org/integration-tests/2019-11-19-050042/

**Comment by ncw** [comment]
I've figured out the problem with this and it should be fixed in 1db31d714907e6720163648fab3175dd5507979e

**Comment by nguyenhuuluan434** [comment]
Sorry i miss your email, thank about taking your time to help me solve this
problem.


Thank you very much,
Luan.

Vào Th 5, 21 thg 11, 2019 vào lúc 20:29 Nick Craig-Wood <
notifications@github.com> đã viết:
